service/storage_service.cc

```diff
@@ -1390,10 +1504,28 @@ std::unordered_set<locator::token> storage_service::get_tokens_for(inet_address
 }
 
 // Runs inside seastar::async context
-void storage_service::set_gossip_tokens(const std::unordered_set<dht::token>& local_tokens) {
+// Assumes that no other functions modify STREAMS_TIMESTAMP, TOKENS or STATUS
+// in the gossiper's local application state while this function runs.
+void storage_service::gossip_joined(
+        const std::unordered_set<dht::token>& tokens, std::optional<db_clock::time_point> streams_ts) {
+    assert(!tokens.empty());
+
+    if (streams_ts) {
```

Comment:
Again, this code can be simplified if we insert STREAMS_TIMESTAMP unconditionally.

Reply:
Done